### PR TITLE
Do not commit text when reset signal is received

### PR DESCRIPTION
**Pull request by epico**
ibus-hangul uses ibus_engine_update_preedit_text_with_mode() to
commit the preedit text by ibus-daemon and now focus-out and
mouse click are handled by ibus-daemon and ibus clients.

BUG=https://github.com/ibus/ibus/issues/1980

### CODE REVIEW

File: src/engine.c [hunk 1]
```diff
@@ -22,6 +22,7 @@
 #include <config.h>
 #endif
 
+#include <stdio.h>
```
Comment (changwoo) [inline src/engine.c]: Minor clean up thing. If this include is for ```atoi()```, the header for ```atoi()``` is ```<stdlib.h>```.  (But I built this commit with removing this line.  Maybe one of the other headers already has it. 🤔)

Comment (epico) [reply src/engine.c]: This is for `sscanf ()`, without this header, gcc 9.1.1 will print out warnings.

Comment (changwoo) [reply src/engine.c]: Sorry I missed the warning. You are right.

It works as intended. 👍

Comment (epico) [reply src/engine.c]: No problem. Thanks for the review!

**Comment by epico** [comment]
Is it okay to merge this patch?

**Comment by choehwanjin** [comment]
Thank you. I've merged this commit.

**Comment by epico** [comment]
Thank you!

**Comment by epico** [comment]
Any idea when you'll release the next version of ibus-hangul?
We plan to update the ibus-hangul package in Fedora.

**Pull request merged**